### DISCUSSION TITLE
DataList control in POS unable to select the Items

### DISCUSSION THREAD

**Discussion by LavanyaNiki** [microsoft/Dynamics365Commerce.Solutions#259] [Q&A]
We have created a new custom form with DataList Control to display a list of information. We select a few things from the Grid, for example, vouchers 1–3 and add them to our transaction.

<img width="929" alt="image" src="https://github.com/microsoft/Dynamics365Commerce.Solutions/assets/140384016/e74e5582-82ac-4455-8188-d65a991850c8">

Then I can go back and check the form to see if I've previously associated the sale. In this case, my previously selected things are unchecked. 

How do I keep the selected items in the datalist?

Followed this documentation:  https://learn.microsoft.com/en-us/dynamics365/commerce/dev-itpro/pos-extension/controls-pos-extension 

and added additional line as below in the OnReady function but no luck.
  this.dataList.selectItems(this.viewModel.voucherNumbers);

Please let us know how to handle this issue?

**Comment by zhangguanghuib** [comment]
I have ever tried select one single item, it is working

**Reply by LavanyaNiki** [reply]
Hello @zhangguanghuib

I tried to select the single item also, but not working.
Is it possible to share the snippet?

**Comment by zhangguanghuib** [comment]
can you please try this:
https://github.com/zhangguanghuib/NewCommerceSDK/blob/main/POS_Samples/POSExtensions/MultiPkgs1/src/ScaleUnitSample/POS/Views/ExampleView.ts

![image](https://github.com/microsoft/Dynamics365Commerce.Solutions/assets/14832260/11592dcb-59c7-401c-a57c-e5fb6f323098)

**Reply by LavanyaNiki** [reply]
@zhangguanghuib : Thank you for the snippet. We tried using this snippet, but it still doesn't work for us.
